lib/cpp/src/thrift/transport/THeaderTransport.cpp

```diff
@@ -394,15 +411,19 @@ void THeaderTransport::setHeader(const string& key, const string& value) {
   writeHeaders_[key] = value;
 }
 
-size_t THeaderTransport::getMaxWriteHeadersSize() const {
+uint32_t THeaderTransport::getMaxWriteHeadersSize() const {
   size_t maxWriteHeadersSize = 0;
   THeaderTransport::StringToStringMap::const_iterator it;
   for (it = writeHeaders_.begin(); it != writeHeaders_.end(); ++it) {
     // add sizes of key and value to maxWriteHeadersSize
     // 2 varints32 + the strings themselves
     maxWriteHeadersSize += 5 + 5 + (it->first).length() + (it->second).length();
   }
-  return maxWriteHeadersSize;
+  if (maxWriteHeadersSize > std::numeric_limits<uint32_t>().max()) {
```

Comment:
Compiling in Win32 exposed additional warnings I will clean up.  Thanks for this comment!  I am also going to build with Visual Studio 2015 in 32-bit and 64-bit to make sure they are all cleaned up.  I have Win64 VS2010 building without any warnings at this point.  In addition I found that the tutorial CMakeLists.txt assumes ZLIB is configured so I changed it to be conditional on the ZLIB library.